byte_mlperf/backends/MIGRAPHX/compile_backend_migraphx.py

```diff
@@ -0,0 +1,153 @@
+        input_tensors = []
+        #convert tf/pytorch model to onnx model, tf works, not verify pytorch now.
+        model_onnx_path=model_path
+        if self.framework == "Tensorflow":
+            model_onnx_path = os.path.join(model_path,self.model_name+ '.onnx')
+            if not os.path.exists(model_onnx_path):
+                saved_to_onnx.savedmodel_to_onnx(model_path, model_onnx_path)
+        elif self.framework == "Pytorch":
+            raise NotImplementedError("MIGraphX backend for models of PyTorch framework has not been implemented yet.")
+        batch_sizes = config['workload']['batch_sizes']
+        for batch_size in batch_sizes:
+            model_path_for_batch_size = model_onnx_path.rsplit("/",1)
+            model_onnx_path_set_batch_size = os.path.join(model_path_for_batch_size[0],str(batch_size),model_path_for_batch_size[1])
+            model_dir = os.path.join(model_path_for_batch_size[0],str(batch_size))
+            if not os.path.exists(model_dir):
+                os.makedirs(model_dir)
+            model_paths = model_onnx_path_set_batch_size.split('.')
+            model_path = model_paths[0] + "_optimized." + model_paths[1]
+            if not os.path.exists(model_path):
+                new_input = {}
+                for key, value in config['model_info']['input_shape'].items():
+                    if(('videobert' in config['model_info']['model']) and (key=='text')):
+                        new_input[key]=value
+                    else:
+                        new_input[key]=[value[0]*batch_size]+value[1:]
+                model = migraphx.parse_onnx(model_onnx_path,map_input_dims=new_input,default_dim_value=batch_size)
+
+                if('FP16' in config['model_info']['model_precision']):
+                    migraphx.quantize_fp16(model, ['dot', 'convolution'])
+
+                model.compile(migraphx.get_target("gpu"))
+                migraphx.save(model, model_path, format='msgpack')
+
+        result = {
+            "model":
+            self.model_name,
+            "framework":
+            config['model_info']['framework'],
+            "compile_precision":
+            config['model_info']['model_precision'],
+            "input_type":
+            self.input_type,
+            "max_batch_size":
+            config['model_info']['max_batch_size'],
+            "compile_status":
+            "success",
+            "sg_percent":
+            100,
+            "segments": [
+                {
+                    "sg_idx":
+                    0,
+                    "is_fallback":
+                    False,
+                    "input_tensor_map":
+                    config['model_info']['input_shape'],
+                    "output_tensor_map":
+                    config['model_info']['outputs'],
+                    "compiled_model": [
+                        {
+                            "compiled_bs": config["workload"]["batch_sizes"][-1],
```

Comment:
In line 75, we compiled multiple batch sizes, so I believe we should have multiple result here as well.

Reply:
Have fixed it. Now we set "compiled_bs" to the list of all batch sizes and "compiled_obj" to the list of saved models' file path.